go/libkb/log_send.go

```diff
@@ -0,0 +1,143 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+import (
+	"bufio"
+	"bytes"
+	"compress/gzip"
+	"mime/multipart"
+	"os"
+	"strings"
+
+	rogReverse "github.com/rogpeppe/rog-go/reverse"
+)
+
+// Logs is the struct to specify the path of log files
+type Logs struct {
+	Desktop string
+	Kbfs    string
+	Service string
+	Updater string
+	Start   string
+}
+
+func addFile(mpart *multipart.Writer, param, filename, data string) error {
+	if len(data) == 0 {
+		return nil
+	}
+
+	part, err := mpart.CreateFormFile(param, filename)
+	if err != nil {
+		return err
+	}
+	gz := gzip.NewWriter(part)
+	if _, err := gz.Write([]byte(data)); err != nil {
+		return err
+	}
+	if err := gz.Close(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Contextified) post(status, kbfsLog, svcLog, desktopLog, updaterLog, startLog string) (string, error) {
```

Comment:
I'd make a new type and not add anything to libkb.Contextified.  That's a special type...